test/notification-notificationcomponent.cpp

```diff
@@ -0,0 +1,490 @@
+/* Icinga 2 | (c) 2025 Icinga GmbH | GPLv2+ */
+
+#include <BoostTestTargetConfig.h>
+#include "base/defer.hpp"
+#include "remote/apilistener.hpp"
+#include "base-testloggerfixture.hpp"
+#include "config/configcompiler.hpp"
+#include "notification/notificationcomponent.hpp"
+
+using namespace icinga;
+
+namespace {
+
+// black magic
+template<auto prvPtr>
+struct InvokeTimerHandlerImpl
+{
+	friend void InvokeTimerHandler(const NotificationComponent::Ptr& nc) { return ((*nc).*prvPtr)(); }
+};
+void InvokeTimerHandler(const NotificationComponent::Ptr& nc);
+
+template struct InvokeTimerHandlerImpl<&NotificationComponent::NotificationTimerHandler>;
+
+} // namespace
+
+class NotificationComponentFixture : public TestLoggerFixture
+{
+public:
+	NotificationComponentFixture()
+	{
+		auto createObjects = []() {
+			String config = R"CONFIG({
+object CheckCommand "dummy" {
+  command = "/bin/echo"
+}
+object Host "h1" {
+  address = "h1"
+  check_command = "dummy"
+  enable_notifications = true
+  enable_active_checks = false
+  enable_passive_checks = true
+}
+object NotificationCommand "send" {
+  command = ["true"]
+}
+apply Notification "n1" to Host {
+  interval = 0
+  command = "send"
+  period = "tp1"
+  users = [ "u1" ]
+  assign where host.enable_notifications == true
+}
+object User "u1" {
+  enable_notifications = true
+}
+object TimePeriod "tp1" {
+  display_name = "Test TimePeriod"
+  ranges = {
+    "monday" = "00:00-24:00"
+	"tuesday" = "00:00-24:00"
+	"wednesday" = "00:00-24:00"
+	"thursday" = "00:00-24:00"
+	"friday" = "00:00-24:00"
+	"saturday" = "00:00-24:00"
+	"sunday" = "00:00-24:00"
+  }
+}
+object NotificationComponent "nc" {}
+})CONFIG";
+			std::unique_ptr<Expression> expr = ConfigCompiler::CompileText("<test>", config);
+			expr->Evaluate(*ScriptFrame::GetCurrentFrame());
+		};
+
+		auto ret = ConfigItem::RunWithActivationContext(new Function("CreateTestObjects", createObjects));
+		BOOST_REQUIRE(ret);
+
+		m_Host = Host::GetByName("h1");
+		BOOST_REQUIRE(m_Host);
+
+		m_Notification = Notification::GetByName("h1!n1");
+		BOOST_REQUIRE(m_Notification);
+
+		m_TimePeriod = TimePeriod::GetByName("tp1");
+		BOOST_REQUIRE(m_TimePeriod);
+
+		ApiListener::UpdateObjectAuthority();
+		BOOST_REQUIRE(ApiListener::UpdatedObjectAuthority());
+
+		// Store the old periods from the config snippets to reuse them later.
+		m_AllTimePeriod = m_TimePeriod->GetRanges();
+
+		Service::OnNotificationSentToUser.connect(NotificationSentToUserHandler);
```

Comment:
Using service to register the signal here looks odd! Why not just use `Checkable` instead?